src/app/members/actions/update-user.ts

```diff
@@ -0,0 +1,19 @@
+"use server";
+import { getSupabaseClient } from "@/utils/supabase/supabaseClient";
+import { UserBuilder } from "../components/MemberFormBuilder";
+import { Tables } from "@/lib/db";
+import { redirect } from "next/navigation";
+import { Routes } from "@/lib/routes";
+
+export const updateUser = async (values: UserBuilder, id: number) => {
+  const supabase = await getSupabaseClient();
+  const { error } = await supabase
```

Comment:
We need to check the user role here to make sure not that only shift incharge and incharge can trigger this endpoint.